striketracker/models/certificate.go

```diff
@@ -0,0 +1,84 @@
+package models
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/openwurl/wurlwind/pkg/validation"
+	validator "gopkg.in/go-playground/validator.v9"
+)
+
+// CertificateResponse encapsulates a list of certificates
+type CertificateResponse struct {
+	List []Certificate `json:"list"`
+}
+
+// Certificate encapsulates a TLS certificate on a subaccount
+type Certificate struct {
+	CABundle               string                  `json:"caBundle"`                        // text of CA bundle
+	Certificate            string                  `json:"certificate" validate:"required"` // text of x.509 cert
+	CertificateInformation *CertificateInformation `json:"certificateInformation"`          // x.509 model
+	Ciphers                string                  `json:"ciphers"`
+	CommonName             string                  `json:"commonName"`
+	CreatedDate            string                  `json:"createdDate"`
+	ExpirationDate         string                  `json:"expirationDate"`
+	Fingerprint            string                  `json:"fingerprint"`
+	ID                     int                     `json:"id"`
+	Issuer                 string                  `json:"issuer"`
+	Key                    string                  `json:"key" validate:"required"`
+	Requester              string                  // awaiting more context
+	Trusted                bool                    `json:"trusted"`
+	UpdatedDate            string                  `json:"updatedDate"`
+}
+
+// Validate validates the Certificate struct data
+func (c *Certificate) Validate() error {
+	v := validation.NewValidator(validator.New())
+	if err := v.Validate(c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CertificateInformation encapsulates a debundled cert
+type CertificateInformation struct {
+	Certificate string `json:"certificate"`
+	Key         string `json:"key"`
+	CABundle    string `json:"caBundle"`
+}
+
+// CertificateHostsUnload is a list of hosts consuming a certificate in its native format
+type CertificateHostsUnload map[string][]*CertificateHost
```

Comment:
That's a mouthful.  Is this a striketracker term?  I don't get the 'unload'

Reply:
The API returns an unmodeled/unstructured response which comes in the format of `map[string][]*Certificatehost`, check out the method attached to it for more clarity on what is being done.

Unload was just a way to differentiate that it is being used to unload data from the API into structs. This struct is only used in the few moments between receiving a response and delivering it to the client requester.

Reply:
A better name may work, but it was quickly done to solve an annoyance

Reply:
also quick clarification that certificates is unfinished its just forklifted into this PR

Reply:
cool, I've use Marshal/Unmarshal in the past.  IMO it has more obvious context of 'moving' data from one format to another.

Reply:
I'm not sure I would like Marshal/Unmarshal since it's usually reserved for actual marshaling operations and may be confusing otherwise. I'll think of something